nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/CSVExportAction.java

```diff
@@ -0,0 +1,309 @@
+            context.askForCheckpoint();
+        }
+
+        // use http://www.algosome.com/articles/how-to-sort-large-file.html
+        protected Blob sort(Blob blob, String commandId) {
+            File file = blob.getFile();
+            if (file == null) {
+                getLog().error(String.format("Blob for %s is not backed by a file, sort is not supported", commandId));
+                return null;
+            }
+            try (Stream<String> lines = Files.lines(file.toPath())) {
+                Path temp = createTemp(commandId);
+                List<String> sorted = lines.sorted().collect(Collectors.toList());
+                Files.write(temp, sorted, StandardOpenOption.WRITE);
+                return new FileBlob(temp.toFile());
+            } catch (IOException e) {
+                getLog().error(e, e);
+                return null;
+            }
+        }
+    }
+
+    public static class ZipBlob extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "zipBlob";
+
+        public ZipBlob() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            try {
+                blob = BlobUtils.zip(blob, null);
+            } catch (IOException e) {
+                getLog().error(e, e);
+            }
+            storeBlob(blob, commandId);
+            context.produceRecord("o1", BulkRecords.of(commandId, countFrom(record), getKey(commandId)));
+            context.askForCheckpoint();
+        }
+
+    }
+
+    public static class ExposeBlobInDownload extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "moveBlobToDownload";
+
+        public ExposeBlobInDownload() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            if (blob != null) {
+                TransientStore download = Framework.getService(TransientStoreService.class)
+                                                   .getStore(DownloadService.TRANSIENT_STORE_STORE_NAME);
+                download.putBlobs(commandId, Collections.singletonList(blob));
+                download.setCompleted(commandId, true);
+            }
+            KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(BULK_KV_STORE_NAME);
+            BulkStatus status = BulkCodecs.getBulkStatusCodec().decode(kvStore.get(commandId + STATUS));
+            BulkCounter counter = new BulkCounter(commandId, status.getCount());
+            context.produceRecord("o1", commandId, BulkCodecs.getBulkCounterCodec().encode(counter));
```

Comment:
What does it change to send the same count to Counter ?

Reply:
already changed, it's now send the COMPLETED state